### PR TITLE
Make distinction between DisplayName and Username in email templates

### LINKED ISSUE #6161: Wrong username in registration notification mail
- Gitea version (or commit ref): 1.7.2
- Git version:  2.20.1.windows.1
- Operating system: Windows 7 x64
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes (provide example URL)
  - [ ] No
  - [ ] Not relevant
- Log gist:

## Description

Upon registration, the notification mail contains :

`You can now login via username: {{.Username}}.`

Where Username is the full name and NOT the real username.

...

### PR DESCRIPTION
**Pull request by mrsdizzie**
Store the actual username in the variable named Username and store the separate DisplayName in another variable. This allows us to access the actual username when we need, which currently fails if a user has set a full name.

Fixes #6161

### CODE REVIEW

File: models/mail.go [hunk 1]
```diff
@@ -47,7 +47,8 @@ func SendTestMail(email string) error {
 // SendUserMail sends a mail to the user
 func SendUserMail(c *macaron.Context, u *User, tpl base.TplName, code, subject, info string) {
 	data := map[string]interface{}{
-		"Username":          u.DisplayName(),
+		"DisplayName":       u.DisplayName(),
+		"Username":          u.DisplayUsername(),
```
Comment (jolheiser) [inline models/mail.go]: This can just be `u.Name`, no need for a func.

File: models/user.go [hunk 2]
```diff
@@ -650,6 +650,11 @@ func (u *User) DisplayName() string {
 	return u.Name
 }
 
+// DisplayUsername returns username
+func (u *User) DisplayUsername() string {
```
Comment (jolheiser) [inline models/user.go]: See other comment.

**Review by jolheiser — CHANGES_REQUESTED** [review]

File: models/mail.go (hunk 1 above)
Comment (mrsdizzie) [reply models/mail.go]: oh yes duh 🙈 fixed

**Comment by jolheiser** [comment]
Do all the maps actually need both `DisplayName` and `Username`? 
It looks like only the `register_notify.tmpl` would need the `Username` whereas the rest use the `DisplayName` exclusively.

**Comment by mrsdizzie** [comment]
True, I guess that is preemptive and if people do want to include the username in the future they can also add it to the map at that time. Appreciate the feedback!

**Review by jolheiser — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Pull request merged**